source/docs/advanced/server/add-http-endpoint.md

```diff
@@ -0,0 +1,107 @@
+
+To do so, we need to declare create our server config file `my-module/server/module.config.js`, and configure its `entrypoint` key.
+
+```js
+module.exports = {
+  endpoint: {
+    // The path added to your Front-Commerce server
+    path: "/wysiwyg-preview",
+    // The router used for this path
+    router: require("./express")
+  }
+};
+```
+
+If we translated this in a standard express application, it would look like this:
+
+```js
+app.use("/wysiwyg-preview", require("./express")())
+```
+
+<blockquote class="note">
+See our express middleware to understand how the magic works: [`withCustomRouters`](https://gitlab.com/front-commerce/front-commerce/blob/85f1a8ef55a351f0feb9309c666992bbbb153993/src/server/express/withCustomRouters.js#L23).
+</blockquote>
+
+Thus, the `my-module/server/express.js` file within your module should be a function returning either a standard express route (`(req, res, next) => { /* ... */ }`) or [an express router](https://expressjs.com/en/api.html#router).
+
+Now, if we want to display our WYSIWYG preview iframe, it could look like this:
+
+```js
+const path = require("path");
+const cors = require("cors");
+const express = require("express");
+const Router = express.Router;
+const config = require("config/website");
+
+const ONE_HOUR = 60 * 60;
+
+module.exports = () => {
+  const router = new Router();
+
+  // We make sure that the resources are only available for
+  // our administration interface
+  router.use(
+    cors({
+      origin: config.admin
+    })
+  );
+
+  // And then, we render a static page that let's you
+  // preview some wysiwyg content
+  router.use(
+    express.static(path.join(process.cwd(), "build/wysiwyg"), {
+      setHeaders: function(res, url, stat) {
+        res.setHeader("Cache-Control", `public, max-age=${ONE_HOUR}`);
+      }
+    })
+  );
+
+  return router;
+};
+```
+
+## Add a global server middleware
+
+We've added our middleware that is now available under the `/wysiwyg-preview` path. But what if we want to add a middleware that should be used before any requests to your application?
+
+To do so, you will need to add the option `__dangerouslyOverrideBasePathChecks` to your endpoint configuration.
+
+```js
+module.exports = {
+  endpoint: {
+    __dangerouslyOverrideBasePathChecks: true,
+    path: "/",
+    router: require("./express")
+  }
+};
+```
+
+Indeed, if you don't set this key to `true`, Front-Commerce will check that you don't setup an invalid path.
+
+For instance, you don't want Front-Commerce to check this if your application is a migration from an older website. Indeed, in this case, you will want to dome some SEO by keeping your old URLs, at least by setting redirections for the ones that don't make sense in your new website. The middleware will let you add this redirection before rendering any page of your website.
```

Comment:
```suggestion
For instance, if your application is a migration from an older website, you might want to create a global middleware and would use `__dangerouslyOverrideBasePathChecks`. For SEO reasons, you may want to keep supporting your old URLs and create redirections to relevant locations in your new website. The middleware will let you add this redirection before rendering any page of your website.
```

Reply:
That's not clearer IMO. I'll try to find a third alternative.

Reply:
I agree :) Thanks.